src/main/java/jsr223/perl/utils/PerlVersionGetter.java

```diff
@@ -55,7 +56,7 @@ public String getPerlVersion(PerlProcessBuilderFactory factory) {
 
         String result = ""; // Empty string for empty result if version recovery fails
 
-        ProcessBuilder processBuilder = factory.getProcessBuilder(PerlPropertyLoader.getInstance().getPerlCommand(),
+        ProcessBuilder processBuilder = factory.getProcessBuilder(PerlCommandCreator.getPerlCommand(),
```

Comment:
as explained before it will not work if PerlCommandCreator.getPerlCommand() is ""
check at least this method on local.